.vitepress/theme/components/SiteMap.vue

```diff
@@ -5,9 +5,8 @@ import { useData } from 'vitepress'
 const data = useData()
 const nav = data.site.value.themeConfig.nav
 const ecosystem = nav.find((i: any) => i.text === 'Ecosystem')
-const items = nav
-  .filter((i: any) => i !== ecosystem && i.items)
-  .concat(ecosystem.items)
+let items = nav.filter((i: any) => i !== ecosystem && i.items)
+items = ecosystem?.items ?  items.concat(ecosystem.items) : items
```

Comment:
I think this change just suppresses the error rather than fixing it.

I believe it's trying to match this entry in the main navigation:

https://github.com/vuejs/docs/blob/e2c54b9b9d66a6e267e42ac55e3098bd46c35eee/.vitepress/config.ts#L38

This fails because it isn't called 'Ecosystem' in translations.

Perhaps it would be better to base the match on `activeMatch` instead, as that should be the same in all languages?

```js
const ecosystem = nav.find((i: any) => i.activeMatch?.includes('ecosystem'))
```

Reply:
Nice, your solution will be better.